sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -193,14 +228,32 @@ private[sql] class JacksonGenerator(
    *
    * @param row The row to convert
    */
-  def write(row: InternalRow): Unit = writeObject(writeFields(row, schema, rootFieldWriters))
+  def write(row: InternalRow): Unit = dataType match {
+    case st: StructType =>
+      writeObject(writeFields(row, st, rootFieldWriters))
+    case _ => throw new UnsupportedOperationException(
```

Comment:
Rethinking about this, I think we can avoid matching the data type and simply do `writeObject(writeFields(row, schema, rootFieldWriters))` like before. When accessing `rootFieldWriters`, there is an exception thrown then.

Reply:
ok. I'll remove it.

Reply:
@viirya I found if we don't check it in here, it will throw `ClassCastExceptoin` from `writeObject()` firstly. So, maybe we don't need to check type in `rootFieldWriters` ?

Reply:
Do you mean `dataType` now is a map type but `writeFields`'s 2nd parameter is a `StructType`?

Reply:
`ClassCastExceptoin` may not informative as `UnsupportedOperationException` here. I'd prefer to see `UnsupportedOperationException`. Let's keep this pattern matching for now.

Reply:
ok got it.

Reply:
I saw you remove the pattern matching?

Reply:
oh I got you wrong. I thought you mean the matching in `rootFieldWeriters`. So we need to keep both of them?

Reply:
hmm, what type of exception thrown now if call `write(row)` when the generator is initialized with `MapType`?

Reply:
It will throw `ClassCastException`.